integration-tests/tests/src/tests/notifications.ts

```diff
@@ -0,0 +1,463 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+class ListObject extends Realm.Object {
+  list!: Realm.List<TestObject>;
+  static schema = {
+    name: "ListObject",
+    properties: {
+      list: { type: "list", objectType: "TestObject" },
+    },
+  };
+}
+
+class TestObject extends Realm.Object {
+  doubleCol!: Realm.Types.Double;
+  static schema = {
+    name: "TestObject",
+    properties: {
+      doubleCol: "double",
+    },
+  };
+}
+
+describe("Notifications", () => {
+  let runCount = 0;
+
+  beforeEach(() => (runCount = 0));
+
+  describe("Realm notifications", () => {
+    openRealmBeforeEach({ schema: [TestObject] });
+
+    it("should fire on change", function (this: RealmContext, done) {
+      this.realm.addListener("change", (sender, event) => {
+        expect(sender).deep.equals(this.realm);
```

Comment:
I guess `sender` in this case is a `realm`?  Perhaps rename `sender` to be more clear.

Also interested to see if `expect(sender).deep.equals({})` passes.

Reply:
I actually wanted to ask about comparing realms as I was not sure how to actually do this. I called it sender because this is what the addListener function calls its callback's parameters, so perhaps it is worth renaming it there too, I did find it confusing.

Reply:
`        expect(sender).deep.equals(this.realm);` does indeed pass... Should I compare their schemas? I think that's what I did originally, I am not sure what's best here.

Reply:
It's not a very valuable assertion in any case.  We register a listener when anything on the realm changes, and then compare if the realm is the same.  Can probably just remove it 🤷🏼